docs/reference.cue

```diff
@@ -9,6 +9,19 @@ _values: {
 	instance:          "vector:9598"
 }
 
+// `#AcknowledgementsStatus` documents the status of end-to-end
+// acknowledgements for a source or sink.
+//
+// * `enabled` - On a source, indicates it will wait for
+// acknowledgements on all incoming events before responding to
+// the request. On a sink, indicates it will mark events with the
+// final delivery status.
+// * `disabled` - Indicates the source or sink does not actively
+// participate in handling acknowledgements.
+// * `not_applicable` - Indicates the source cannot acknowledge
+// the transmission of data.
+#AcknowledgementsStatus: "enabled" | "disabled" | "not_applicable"
+
```

Comment:
After thinking about this more, I'd prefer that we unify`#AcknowledgementsStatus` and `#DeliveryStatus`. I’m of the mindset that a component is `at_least_once` possible if it has both:

1. An underlying protocol that acknowledges receipt
2. Implements Vector-level acknowledgements

If either are violated then the component is `best_effort`, otherwise the guarantee is satisfied from a user's perspective.

cc @lucperkins

Reply:
It seems somewhat reductive to boil it all down to just two states, but I think I see what you're getting at. This will of course take the `at_least_once` state away from most of the components (for now), but that may well have been misleading anyways.

Reply:
How should this work for sources that scrape metrics? They can't really be `at_least_once`, but at least some of the sources like this are marked that way.